x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/alert_create_flyout.ts

```diff
@@ -85,6 +85,41 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       await testSubjects.click('rulesTab');
     });
 
+    it('should delete the right action when the same action has been added twice', async () => {
+      // create a new rule
+      const ruleName = generateUniqueKey();
+      await rules.common.defineIndexThresholdAlert(ruleName);
+
+      // create webhook connector
+      await testSubjects.click('.webhook-alerting-ActionTypeSelectOption');
+      await testSubjects.click('createActionConnectorButton-0');
+      await testSubjects.setValue('nameInput', 'webhook-test');
+      await testSubjects.setValue('webhookUrlText', 'https://test.test');
+      await testSubjects.setValue('webhookUserInput', 'fakeuser');
+      await testSubjects.setValue('webhookPasswordInput', 'fakepassword');
+
+      // save rule
+      await find.clickByCssSelector('[data-test-subj="saveActionButtonModal"]:not(disabled)');
+      await find.setValueByClass('kibanaCodeEditor', 'myUniqueKey');
+      await testSubjects.click('saveRuleButton');
+
+      // add new action and remove first one
+      await testSubjects.click('ruleSidebarEditAction');
+      await testSubjects.click('.webhook-alerting-ActionTypeSelectOption');
+      await find.clickByCssSelector(
+        '[data-test-subj="alertActionAccordion-0"] [aria-label="Delete"]'
+      );
+
+      // check that the removed action is the right one
+      const doesExist = await find.existsByXpath(".//*[text()='myUniqueKey']");
```

Comment:
Couldn't think of a better way to check that the right action is removed. Open to suggestions

Reply:
In general, we try to use test subjects over xpath expressions where possible. In this case, the xpath way seems reasonable to me.